### PR TITLE
Remove early reference to "error" in tail call example

### LINKED ISSUE #183: The "error" in factorial function in Methods chapter is not described
It would be nice to have a reference to error handling docs after the factorial function example. Without it the "if x < 0 then error end" line looks like error :-).

### PR DESCRIPTION
**Pull request by SeanTAllen**
Closes #183

### CODE REVIEW

File: expressions/methods.md [hunk 1]
```diff
@@ -32,13 +32,15 @@ If you want to exit a function early then use the `return` command. If the funct
...
 ```pony
-fun factorial(x: I32): I32 ? =>
-  if x < 0 then error end
-  if x == 0 then
-    1
-  else
-    x * factorial(x - 1)?
-  end
+primitive Factorial
+  fun factorial(x: I32): I32 ? =>
```
Comment (aturley) [inline expressions/methods.md]: Since the function can no longer raise an error it no longer needs to be partial.

**Comment by SeanTAllen** [comment]
odd that it compiles.

**Comment by SeanTAllen** [comment]
@aturley updated.

**Comment by aturley** [comment]
@SeanTAllen I filed #2396 in response to the fact that your partial version compiled even though the function wasn't really partial.

**Comment by SeanTAllen** [comment]
@aturley i had already opened an issue for that and closed #2396 as a duplicate.

**Comment by SeanTAllen** [comment]
@jemc updated to U32. merging.

**Comment by SeanTAllen** [comment]
Apparently I forgot to actually push the U32 change. need to wake up more.

**Comment by SeanTAllen** [comment]
Fixing on master.

**Pull request merged**